control-plane/agents/ha-node/src/server.rs

```diff
@@ -0,0 +1,261 @@
+use crate::{
+    detector::{NvmeController, NvmePathCache},
+    path_provider::get_nvme_path_buf,
+};
+use anyhow::anyhow;
+use common_lib::transport_api::{ReplyError, ResourceKind};
+use grpc::operations::ha_node::{
+    server::NodeAgentServer,
+    traits::{NodeAgentOperations, ReplacePathInfo},
+};
+use http::Uri;
+use nvmeadm::{
+    nvmf_discovery::{ConnectArgs, ConnectArgsBuilder},
+    nvmf_subsystem::Subsystem,
+};
+use std::{net::SocketAddr, sync::Arc};
+use tokio::time::{sleep, Duration};
+use tonic::transport::Server;
+use utils::NVME_TARGET_NQN_PREFIX;
+
+/// Common error source name for all gRPC errors in HA Node agent.
+const HA_AGENT_ERR_SOURCE: &str = "HA Node agent gRPC server";
+
+/// NVMe subsystem refresh interval when monitoring its state, in milliseconds.
+const SUBSYSTEM_STATE_REFRESH_INTERVAL_MS: u64 = 500;
+
+/// TODO:
+pub struct NodeAgentApiServer {
+    endpoint: SocketAddr,
+    path_cache: NvmePathCache,
+}
+
+impl NodeAgentApiServer {
+    pub fn new(uri: Uri, path_cache: NvmePathCache) -> Self {
+        let endpoint = uri.authority().unwrap().to_string().parse().unwrap();
+
+        Self {
+            endpoint,
+            path_cache,
+        }
+    }
+
+    pub async fn start(&self) -> anyhow::Result<()> {
```

Comment:
nit: would serve or run be a better name?

Reply:
I don't mind, actually. Will change it to 'serve()'.